--- a/Makefile	2024-02-16 03:04:21.309522629 +0000
+++ b/Makefile	2024-02-16 03:05:38.626046427 +0000
@@ -997,6 +997,8 @@
 # Avoid 'Not enough room for program headers' error on binutils 2.28 onwards.
 LDFLAGS_u-boot += $(call ld-option, --no-dynamic-linker)
 
+LDFLAGS_u-boot += -z notext $(call ld-option, --apply-dynamic-relocs)
+
 LDFLAGS_u-boot += --build-id=none
 
 ifeq ($(CONFIG_ARC)$(CONFIG_NIOS2)$(CONFIG_X86)$(CONFIG_XTENSA),)
